src/optional-listeners/message-command-listeners/CorePreMessageParser.ts

```diff
@@ -42,11 +42,10 @@ export class CoreListener extends Listener<typeof Events.PreMessageParsed> {
 	private async canRunInChannel(message: Message): Promise<boolean> {
 		if (isDMChannel(message.channel)) return true;
 
-		const me = message.guild!.me ?? (message.client.id ? await message.guild!.members.fetch(message.client.id) : null);
+		const me = await message.guild?.members.fetchMe();
```

Comment:
Rather than `me` potentially being sync if available it is now always async. I'd rather prefer another solution when it can still be sync if available.

Reply:
Wouldn't matter as if the member is cached its immediately returned, and the entire method is async anyways